app/controllers/admin/connected_apps_controller.rb

```diff
@@ -26,6 +21,45 @@ def destroy
 
     private
 
+    def create_connected_app
+      attributes = {}
+      attributes[:type] = connected_app_params[:type] if connected_app_params[:type]
+
+      @app = ConnectedApp.create!(enterprise_id: enterprise.id, **attributes)
+    end
+
+    def connect
+      return connect_vine if connected_app_params[:type] == "ConnectedApps::Vine"
+
+      create_connected_app
+      @app.connect(api_key: spree_current_user.spree_api_key,
+                   channel: SessionChannel.for_request(request))
+    end
+
+    def connect_vine
```

Comment:
I think that we can leave it for now but it would be better to move app-specific logic into the apps and have the controller be agnostic of the app's needs.

Reply:
To be honest I would prefer to have the connect logic to be separated from the model and having a service handle the connection bit before storing the result in the model.
But yeah, this should live in a service.

Reply:
Good point, too. If we take the logic out of the models then we don't really need a polymorphic model. The beauty of the polymorphic model is that it loads the right code automatically. But reducing models to data storage is really handy as well. Then we just need service objects that are coupled to app types. I'm not sure which is better. Maybe it's just that the VINE app has some additional logic that should go into a service.

Reply:
I feel like we might need other connected apps to see if a pattern emerges before we can refine the design. Maybe VINE is the odd one who knows. If I get a bit of time, I'll look at moving the VINE logic to a service.